### PR TITLE
[DOCS] Clarifying tiers order in rolling_upgrade

### PR DESCRIPTION
**Pull request by Leaf-Lin**
A follow up on #77491 to add node attributes comment in the upgrade order.

### CODE REVIEW

**Review by DaveCTurner — COMMENTED** [review]
I left a couple of suggestions; could you reformat to fit in 80 columns too?

**Review by DaveCTurner — COMMENTED** [review]
Oh also please mention somewhere in the commit message that this is a follow-up to #77491.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 1]
```diff
@@ -11,10 +11,9 @@ We strongly recommend that when you upgrade you divide your cluster's nodes
...
 
 . Nodes that are not <<master-node,master-eligible>>. You can retrieve a list
-of these nodes with `GET /_nodes/_all,master:false` or by finding all the nodes
+of these nodes with `GET /_nodes/_all,master:false/_none` or by finding all the nodes
 configured with `node.master: false`.
-.. If you are using data tiers, you should
-upgrade the nodes by tier, completing one tier at a time, in this order: frozen, cold, warm and finally hot tier.
+.. If you are using data tiers, or a hot-warm-cold architecture based on node attributes, you should upgrade the nodes tier-by-tier, completing the upgrade of each tier before starting the next. Upgrade the frozen tier first, then the cold tier, then the warm tier, and finally the hot tier.
 This is to ensure ILM can continue to move between the phases and ensure version
 compatibility.
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply docs/reference/upgrade/rolling_upgrade.asciidoc:18]: :+1: although without the paragraph break.

**Review by DaveCTurner — COMMENTED** [review]
Sorry for the delay here. Seems like a good change to me. Could you reformat the source to be 80 columns wide?

**Review by Leaf-Lin — COMMENTED** [review]
I think this is the expected format and paragraph break.

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 2]
```diff
@@ -11,14 +11,19 @@ We strongly recommend that when you upgrade you divide your cluster's nodes
...
+of these nodes with `GET /_nodes/_all,master:false/_none` or by finding all the
+nodes configured with `node.master: false`.
+
+.. If you are using data tiers, or a hot-warm-cold architecture based on node 
+attributes, you should upgrade the nodes tier-by-tier, completing the upgrade
+of each tier before starting the next. Upgrade the frozen tier first, then the
+cold tier, then the warm tier, and finally the hot tier.  This is to ensure ILM
+can continue to move between the phases and ensure version
+compatibility. You can get the list of nodes from specific tiers by `GET 
+/_nodes/data_frozen:true/_none`, `GET /_nodes/data_cold:true/_none` etc.
```
Comment (DaveCTurner) [inline docs/reference/upgrade/rolling_upgrade.asciidoc:23]: Unfortunately a line break in the middle of `preformatted text` gets copied into the output:

image

It needs to be this instead:

```suggestion
compatibility. You can get the list of nodes from specific tiers by
`GET /_nodes/data_frozen:true/_none`, `GET /_nodes/data_cold:true/_none` etc.
```

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 3]
```diff
@@ -11,14 +11,19 @@ We strongly recommend that when you upgrade you divide your cluster's nodes
...
-.. If you are not using data tiers, you may upgrade the nodes within the group in any order.
-
+of these nodes with `GET /_nodes/_all,master:false/_none` or by finding all the
+nodes configured with `node.master: false`.
+
+.. If you are using data tiers, or a hot-warm-cold architecture based on node 
+attributes, you should upgrade the nodes tier-by-tier, completing the upgrade
+of each tier before starting the next. Upgrade the frozen tier first, then the
+cold tier, then the warm tier, and finally the hot tier.  This is to ensure ILM
+can continue to move between the phases and ensure version
```
Comment (DaveCTurner) [inline docs/reference/upgrade/rolling_upgrade.asciidoc:21]:
```suggestion
can continue to move shards between phases and ensure version
```

**Review by DaveCTurner — APPROVED** [review]
A couple more suggestions, but otherwise this LGTM.

**Pull request merged**